python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/dagster_sling_translator.py

```diff
@@ -0,0 +1,196 @@
+import re
+from typing import Any, Iterable, Mapping, Optional
+
+from dagster import (
+    AssetKey,
+    AutoMaterializePolicy,
+    FreshnessPolicy,
+    JsonMetadataValue,
+    MetadataValue,
+)
+from dagster._annotations import public
+
+
+class DagsterSlingTranslator:
+    def __init__(self, target_prefix="target"):
+        """A class that allows customization of how to map a Sling stream to a Dagster AssetKey.
+
+        Args:
+            target_prefix (str): The prefix to use when creating the AssetKey
+        """
+        self.target_prefix = target_prefix
+
+    @public
+    @classmethod
+    def sanitize_stream_name(cls, stream_name: str) -> str:
```

Comment:
Purely for my own knowledge, what are cases when a user would want to override this method?

Reply:
Only ones I could come up with are that 1) they don't want to replace with an underscore, but maybe remove it altogether, or 2) lowercase everything.